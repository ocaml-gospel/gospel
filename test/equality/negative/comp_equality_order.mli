type ('a, 'b) t

val f :
  ('b -> 'b -> int) -> ('a -> 'a -> int) -> ('a, 'b) t -> ('a, 'b) t -> int
(*@ comparison *)
