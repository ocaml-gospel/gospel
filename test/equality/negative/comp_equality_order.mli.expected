
*******************************
********** Parsed file ********
*******************************
type ('a, 'b) t
val f :
  ('b -> 'b -> int) -> ('a -> 'a -> int) -> ('a, 'b) t -> ('a, 'b) t -> int
[@@gospel {| comparison |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

type ('a, 'b) t
  

val f :
  ('b -> 'b -> int) -> ('a -> 'a -> int) -> ('a, 'b) t -> ('a, 'b) t -> int
(*@ comparison
     *)
File "comp_equality_order.mli", lines 3-5, characters 0-27:
Error: Type checking error: type is incompatible with equality clause
